File: idro-frontend/remove_comments.py
import re
import os
import sys

def remove_comments(text, ext):
    if ext in ['.js', '.jsx']:
        # Step 1: Remove JSX context comments {/* ... */}
        pattern_jsx = r'\{\s*/\*[\s\S]*?\*/\s*\}'
        text = re.sub(pattern_jsx, "", text, flags=re.MULTILINE)

        # Step 2: Regular JS strings/comments
        pattern = r'("(?:[^"\\]|\\.)*"|\'(?:[^\'\\]|\\.)*\'|`(?:[^`\\]|\\.)*`)|(/\*[\s\S]*?\*/|\/\/.*$)'
        def replacer(match):
            if match.group(1): # It's a string
                return match.group(1)
            else: # It's a comment
                return ""
        return re.sub(pattern, replacer, text, flags=re.MULTILINE)
    elif ext == '.css':
        # CSS only has /* */ comments
        pattern = r"((['\"])(?:(?!\2).|\\.)*\2)|(/\*[\s\S]*?\*/)"
        def replacer(match):
            if match.group(1):
                return match.group(1)
            else:
                return ""
        return re.sub(pattern, replacer, text, flags=re.MULTILINE)
    return text

def process_file(filepath):
    _, ext = os.path.splitext(filepath)
    if ext not in ['.js', '.jsx', '.css']:
        return

    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()

        new_content = remove_comments(content, ext)
        
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Processed: {filepath}")
    except Exception as e:
        print(f"Error processing {filepath}: {e}")

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python remove_comments.py <directory_or_file>")
        sys.exit(1)

    target = sys.argv[1]
    if os.path.isfile(target):
        process_file(target)
    elif os.path.isdir(target):
        for root, dirs, files in os.walk(target):
            for file in files:
                process_file(os.path.join(root, file))

File: idro-frontend/removeComments.js
const fs = require('fs');
const path = require('path');

function removeComments(text, ext) {
    if (ext === '.js' || ext === '.jsx') {
        // Step 1: Remove JSX context comments {/* ... */}
        const jsxPattern = /\{\s*\/\*[\s\S]*?\*\/\s*\}/g;
        text = text.replace(jsxPattern, "");

        // Step 2: Regular JS strings/comments
        const jsPattern = /("(?:[^"\\]|\\.)*"|'(?:[^'\\]|\\.)*\'|`(?:[^`\\$]|\\.|(\$\{[^}]*\}))*`|\/\*[\s\S]*?\*\/|\/\/.*)/g;
        
        return text.replace(jsPattern, (match) => {
            if (match.startsWith('//') || match.startsWith('/*')) {
                return "";
            }
            return match;
        });
    } else if (ext === '.css') {
        // CSS only has /* */ comments
        const cssPattern = /("(?:[^"\\]|\\.)*"|'(?:[^'\\]|\\.)*'|\/\*[\s\S]*?\*\/)/g;
        return text.replace(cssPattern, (match) => {
            if (match.startsWith('/*')) return "";
            return match;
        });
    }
    return text;
}

function processFile(filepath) {
    const ext = path.extname(filepath);
    if (!['.js', '.jsx', '.css'].includes(ext)) return;

    try {
        const content = fs.readFileSync(filepath, 'utf8');
        const newContent = removeComments(content, ext);
        fs.writeFileSync(filepath, newContent, 'utf8');
        console.log(`Processed: ${filepath}`);
    } catch (e) {
        console.error(`Error processing ${filepath}: ${e.message}`);
    }
}

function walk(dir) {
    const files = fs.readdirSync(dir);
    files.forEach(file => {
        const filepath = path.join(dir, file);
        const stats = fs.statSync(filepath);
        if (stats.isDirectory()) {
            walk(filepath);
        } else {
            processFile(filepath);
        }
    });
}

const logFile = path.join(__dirname, 'log.txt');
function log(msg) {
    fs.appendFileSync(logFile, msg + '\n');
    console.log(msg);
}

const target = process.argv[2];
if (!target) {
    log("Usage: node removeComments.js <directory_or_file>");
    process.exit(1);
}

// Self-test
const testCode = "const x = 1; // comment\n/* multiline */\nconst y = 'http://url';";
const cleaned = removeComments(testCode, '.js');
log("Self-test cleaned result:\n" + cleaned);

try {
    const stats = fs.statSync(target);
    if (stats.isFile()) {
        processFile(target);
    } else if (stats.isDirectory()) {
        walk(target);
    }
} catch (e) {
    log(`Error: ${e.message}`);
}
